[Jonas Ng Zuo En] iP

jonasngs opened the pull request.

Commit:
Add Gradle support

jonasngs committed:
added greetings and echo feature

jonasngs committed:
Added a todo list and features to add items into the list and display it

jonasngs committed:
Created a Task class and a feature to mark tasks as done

jonasngs committed:
Added Deadline, Todo and Event class as well as made changes to Duke class

jonasngs committed:
Added Automatic Text UI Testing

jonasngs committed:
Added DukeException class and respective sub classes to handle incorrect inputs

jonasngs committed:
Added a function to delete tasks

jonasngs committed:
Added more inputs for Automated Text UI testing

jonasngs committed:
Added word art for duke greeting and modified error handling

jonasngs committed:
Modified Duke to account for minor bugs and error

jonasngs committed:
Added more inputs for automated text UI testing

jonasngs committed:
Modified Duke.js for error handling

jonasngs committed:
Changed error messages for invalid task creation

jonasngs committed:
Added more inputs in automated text UI test to account for empty description during task creation

jonasngs committed:
Change indentation for greeting string

jonasngs committed:
Handle more exceptions for missing inputs

jonasngs committed:
Implement Level-7

jonasngs committed:
Implement level 8

jonasngs committed:
Merge branch 'branch-Level-7'

jonasngs committed:
Merge branch-Level-8 to master branch

jonasngs committed:
Modify retrieveTaskInfo method

jonasngs committed:
Handle errors for empty description and error regarding reading of tasks from file

jonasngs committed:
Refactor code to adhere to OOP

jonasngs committed:
Organize classes into packages

jonasngs committed:
Add inputs for Automated text UI test

jonasngs committed:
Add JUnit tests

jonasngs committed:
Add A-Jar

jonasngs committed:
Implement A-JavaDocs

jonasngs committed:
Merge branch 'branch-A-JavaDoc'

jonasngs committed:
Modify code to adhere to coding standards

jonasngs committed:
Merge branch 'branch-A-CodingStandard'

jonasngs committed:
Implement Level-9 to add find function

jonasngs committed:
Merge branch 'branch-Level-9'

jonasngs committed:
Fix a bug when searching for tasks by date

jonasngs committed:
Modify find function to be case insensitive

jonasngs committed:
Add an InvalidCommand class to handle invalid commands by users

jonasngs committed:
Add completeTask method in TaskList class for usage in CompleteTaskCommand class

jonasngs committed:
Organise import statements and remove unnecessary import statements

jonasngs committed:
Change access modifier of instance fields to private and implement getter methods

jonasngs committed:
Modify code to adhere to coding standards

jonasngs committed:
Add javadocs for class variables

jonasngs committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

jonasngs committed:
Add gradle support

jonasngs committed:
Merge branch 'branch-A-Gradle'

jonasngs committed:
Add checkstyle

jonasngs committed:
Merge branch 'branch-A-CheckStyle'

jonasngs committed:
Edit code to comply with checkstyle

jonasngs committed:
Add JavaFX dependencies to build.gradle

jonasngs committed:
Create basic GUI for duke using JavaFX

-Add feature to echo user
-Add dialog boxes
-Add method to flip dialog box to differentiate user and duke output
-Rebuild scene using FXML

TanLeYang commented on src/main/java/duke/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,123 @@
+
+import duke.Storage;
+import duke.Ui;
+import duke.exception.DukeException;
```
Perhaps you could consider leaving some blank lines in between imports from different sub-packages to improve readability! Noticed that you did so in some cases but not all. Maybe it would be better to be consistent throughout!

TanLeYang commented on src/main/java/duke/command/FindByDateCommand.java:
```diff
@@ -0,0 +1,80 @@
+            ui.printReply("Search Results:");
+            for (Task task : tasks.getTaskList()) {
+                if (task.getDate() != null) {
+                    if (task.getDate().isEqual(dateToSearch)) {
```
Could this nesting of the if statement be replaced with a && operator?

TanLeYang reviewed (commented):
Nice job! Good compliance with the coding standard. I like how you included JavaDoc for almost every public method 👍

jonasngs committed:
Implement getResponse method in Duke class

GUI requires a getResponse method to display the Duke response.

The execute method in Command class does not return a String.

Let's modify the execute method of the Command class such that it
returns the Duke response as a String.

jonasngs committed:
Update GUI for Duke to display welcome message upon launch

jonasngs commented:
> Nice job! Good compliance with the coding standard. I like how you included JavaDoc for almost every public method 👍

Thank you for taking time to review my PR!

jonasngs committed:
Modify GUI components

The previous GUI looked dull and text can be difficult to read.

This can affect user experience and user friendliness.

Let's
 * Add appropriate padding to distinguish user and duke dialog
 * Add a background color for the dialog box to make text easily
   readable.
 * Add padding between each ImageView and its Label in the dialog box.

jonasngs committed:
Fix merge conflicts between master and branch-Level-10

jonasngs committed:
Modify some class names to adhere to coding standards

jonasngs committed:
Modify Ui class to handle printing of responses

Printing of responses are currently handled by the execute method in
command classes.

This violates the single-responsibility principle.

Let's shift all printing of responses functionality to the Ui class
which deals with user interactions.

jonasngs committed:
Modify code to adhere to Java coding standard

jonasngs committed:
Modify dialog box of GUI to distinguish between user text and duke text

jonasngs committed:
Modify the MainWindow component to customise the GUI

jonasngs committed:
Modify GUI to close when the exit command is invoked

The GUI remains open despite the exit command being invoked by the user.

Users can continue typing commands even after the exit command is given.

Let's implement a feature to automatically close the GUI after users
invoke the exit command.

luciatirta commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,77 @@
+ */
+public class Parser {
+
+    private static final String LIST_COMMAND = "list";
```
Could perhaps consider using Enums for the type of commands

luciatirta commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,163 @@
+     * @param listSize Current number of tasks in the user's list of tasks.
+     * @return String containing a reply for successful task creation.
+     */
+    public String printNewTask(Task task, int listSize) {
```
The function name printNewTask() seems like a void function. Since it returns the string representation of the new task, maybe can consider newTaskToString(). Same comment with other similar functions in this file.

luciatirta commented on src/test/java/CommandTest.java:
```diff
@@ -0,0 +1,82 @@
+
+import java.util.ArrayList;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Static import statement should be placed above other import statements

luciatirta reviewed (commented):
Looks good to merge!
The codes are very well structured with OOP principles and comprehensive JavaDocs.
Very clear exception messages and commands list for great user experience.
Minor improvements on the function naming can be made. 
Keep up the good work!

jonasngs commented:
> Looks good to merge!
> The codes are very well structured with OOP principles and comprehensive JavaDocs.
> Very clear exception messages and commands list for great user experience.
> Minor improvements on the function naming can be made.
> Keep up the good work!

Thank you for taking time to review my PR!

jonasngs committed:
Add Junit tests for Command and Task classes

jonasngs committed:
Modify method names in Ui class to adhere to coding standards

jonasngs committed:
Implement A-Assertions

jonasngs committed:
Implement A-CodeQuality

The storage and command classes have methods that do not adhere to good
code quality practices.

This can cause the code to become less readable and understandable.

Let's modify the problematic methods to improve code quality.

jonasngs committed:
Implement A-Lambdas

jonasngs committed:
Implement A-Streams

jonasngs committed:
Merge pull request #1 from jonasngs/branch-A-Assertions

Implement A-Assertions

jonasngs committed:
Merge branch 'master' into branch-A-CodeQuality

jonasngs committed:
Merge branch 'master' into branch-A-Lambdas

jonasngs committed:
Merge branch 'master' into branch-A-Streams

jonasngs committed:
Merge pull request #2 from jonasngs/branch-A-CodeQuality

Implement A-CodeQuality

jonasngs committed:
Merge branch 'master' into branch-A-Lambdas

jonasngs committed:
Merge branch 'master' into branch-A-Streams

jonasngs committed:
Merge pull request #3 from jonasngs/branch-A-Lambdas

Implement A-Lambdas

jonasngs committed:
Merge branch 'master' into branch-A-Streams

jonasngs committed:
Merge pull request #4 from jonasngs/branch-A-Streams

Implement A-Streams

jonasngs committed:
Add C-Tagging

jonasngs committed:
Merge branch 'branch-C-Tagging'

jonasngs committed:
Modify Duke GUI

jonasngs committed:
Add HelpCommand class

jonasngs committed:
Modify code to adhere to Java coding standard

jonasngs committed:
Add missing javadocs for GUI related classes

jonasngs committed:
Add A-CI

jonasngs committed:
Merge branch 'branch-A-CI'

jonasngs committed:
Rearrange import statements to adhere to checkstyle requirements

jonasngs committed:
Add Junit tests for Command classes

jonasngs committed:
Modify Duke GUI to minimize the spacing between dialog boxes

jonasngs committed:
Shift parsing functionalities to Parser class

The command classes currently perform additional parsing of user
input to retrieve the necessary information for command execution.

This is not the ideal implementation since all parsing of user inputs
should be handled by the Parser class.

Let's abstract all parsing methods in the Command classes to the Parse
class.

jonasngs committed:
Modify javadocs to adhere to Java coding standards

jonasngs committed:
Add A-BetterGui

jonasngs committed:
Resolve checkstyle issues

jonasngs committed:
Add representative screenshot

jonasngs committed:
Update README.md

jonasngs committed:
Add A-UserGuide

jonasngs committed:
Set theme jekyll-theme-cayman

jonasngs committed:
Rename Ui.png file

jonasngs committed:
Merge branch 'master' of https://github.com/jonasngs/ip

jonasngs committed:
Resolve file naming issue for Ui.png

jonasngs committed:
Add representative screenshot

jonasngs committed:
Update README.md

jonasngs committed:
Update README.md

jonasngs committed:
Update README.md

jonasngs committed:
Update README.md
